GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -232,84 +426,193 @@ namespace gdjs {
      */
     private _storeSoundInArray(arr: Array<HowlerSound>, sound): any {
       //Try to recycle an old sound.
-      for (let i = 0, len = arr.length; i < len; ++i) {
-        if (arr[i] !== null && arr[i].canBeDestroyed()) {
+      for (var i = 0, len = arr.length; i < len; ++i) {
+        if (arr[i] !== null && arr[i].stopped()) {
           arr[i] = sound;
           return sound;
         }
       }
+
       arr.push(sound);
       return sound;
     }
 
-    playSound(soundName, loop, volume, pitch) {
+    /**
+     * Creates a new gdjs.HowlerSound using preloaded/cached Howl instances.
+     * @param soundName The name of the file or resource to play.
+     * @param isMusic True if a music, false if a sound.
+     */
+    createHowlerSound(soundName: string, isMusic: boolean): HowlerSound {
       const soundFile = this._getFileFromSoundName(soundName);
-      const sound = new gdjs.HowlerSound({
+      const cacheContainer = isMusic ? this._loadedMusics : this._loadedSounds;
+
+      if (!cacheContainer.hasOwnProperty(soundFile)) {
+        //@ts-ignore FIXME Somehow TypeScript does not detect the Howl consturctor
+        cacheContainer[soundFile] = new Howl({
+          src: [soundFile],
+          preload: true,
+          html5: isMusic,
+          onplayerror: (_, error) =>
+            console.error("Can't play a sound. Error is: ", error),
```

Comment:
`Can't play an audio file` - to avoid confusion between GDevelop music and sounds distinction?

Reply:
Well, it's true that the manager itself is called SoundManager and Howler uses "HowlerSound". Might be still be good to be precise, either "audio file" or "can't play a sound with Howler"?